skorch/callbacks/lr_scheduler.py

```diff
@@ -140,3 +162,192 @@ def get_lr(self):
+      or each group (list). Functionally, it defines the cycle
+      amplitude (max_lr - base_lr). The lr at any cycle is the sum
+      of base_lr and some scaling of the amplitude; therefore max_lr
+      may not actually be reached depending on scaling function.
+
+    step_size : int (default=2000)
+      Number of training iterations per half cycle. Authors suggest
+      setting step_size 2-8 x training iterations in epoch.
+
+    mode : str (default='triangular')
+      One of {triangular, triangular2, exp_range}. Values correspond
+      to policies detailed above. If scale_fn is not None, this
+      argument is ignored.
+
+    gamma : float (default=1.0)
+      Constant in 'exp_range' scaling function:
+      gamma**(cycle iterations)
+
+    scale_fn : function (default=None)
+      Custom scaling policy defined by a single argument lambda
+      function, where 0 <= scale_fn(x) <= 1 for all x >= 0.
+      mode paramater is ignored.
+
+    scale_mode : str (default='cycle')
+      One of {'cycle', 'iterations'}. Defines whether scale_fn
+      is evaluated on cycle number or cycle iterations (training
+      iterations since start of cycle).
+
+    last_batch_idx : int (default=-1)
+      The index of the last batch.
+
+    Examples
+    --------
+
+    >>> optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9)
+    >>> scheduler = torch.optim.CyclicLR(optimizer)
+    >>> data_loader = torch.utils.data.DataLoader(...)
+    >>> for epoch in range(10):
+    >>>     for batch in data_loader:
+    >>>         scheduler.batch_step()
+    >>>         train_batch(...)
+
+    References
+    ----------
+    .. [1] Leslie N. Smith, 2017, "Cyclical Learning Rates for Training Neural Networks,". "ICLR" `<https://arxiv.org/abs/1506.01186>`_
+
+    """
+
+    def __init__(self, optimizer, base_lr=1e-3, max_lr=6e-3,
+                 step_size=2000, mode='triangular', gamma=1.,
+                 scale_fn=None, scale_mode='cycle',
+                 last_batch_idx=-1):
+
+        if not isinstance(optimizer, Optimizer):
+            raise TypeError('{} is not an Optimizer'.format(
+                type(optimizer).__name__))
+        self.optimizer = optimizer
+
+        if isinstance(base_lr, (list, tuple)):
+            if len(base_lr) != len(optimizer.param_groups):
+                raise ValueError("expected {} base_lrs, got {}".format(
+                    len(optimizer.param_groups), len(base_lr)))
+            self.base_lrs = list(base_lr)
+        else:
+            self.base_lrs = [base_lr] * len(optimizer.param_groups)
+
+        if isinstance(max_lr, (list, tuple)):
+            if len(max_lr) != len(optimizer.param_groups):
+                raise ValueError("expected {} max_lrs, got {}".format(
+                    len(optimizer.param_groups), len(max_lr)))
+            self.max_lrs = list(max_lr)
+        else:
+            self.max_lrs = [max_lr] * len(optimizer.param_groups)
```

Comment:
The `base_lr` and `max_lr` initialization could be refactored in a similar way to `WarmRestartLR._format_lr` but we can do this later.